pkg/consensus/client.go

```diff
@@ -49,36 +50,47 @@ func NewClient(endpoint string, clock *Clock, logger *zap.Logger) *Client {
 		},
 		Codec: eth2api.JSONCodec{},
 	}
+	proposerCache, err := lru.New(cacheSize)
+	if err != nil {
+		return &Client{}, err
+	}
+	executionHash, err := lru.New(cacheSize)
+	if err != nil {
+		return &Client{}, err
+	}
 	return &Client{
 		logger:         logger,
 		client:         client,
 		clock:          clock,
-		proposerCache:  make(map[types.Slot]ValidatorInfo),
-		executionCache: make(map[types.Slot]types.Hash),
-	}
+		proposerCache:  *proposerCache,
+		executionCache: *executionHash,
+	}, nil
 }
 
 func (c *Client) GetParentHash(slot types.Slot) (types.Hash, error) {
 	targetSlot := slot - 1
-	c.executionCacheMutex.RLock()
-	parentHash, ok := c.executionCache[targetSlot]
-	c.executionCacheMutex.RUnlock()
+	parentHash, ok := c.executionCache.Get(targetSlot)
 	if !ok {
 		return c.fetchExecutionHash(targetSlot)
 	}
-	return parentHash, nil
+	if hash, ok := parentHash.(types.Hash); ok {
+		return hash, nil
+	} else {
+		return types.Hash{}, fmt.Errorf("invalid value stored in cache")
+	}
 }
 
 func (c *Client) GetProposerPublicKey(slot types.Slot) (*types.PublicKey, error) {
-	c.proposerCacheMutex.RLock()
-	defer c.proposerCacheMutex.RUnlock()
-
-	validator, ok := c.proposerCache[slot]
+	validator, ok := c.proposerCache.Get(slot)
 	if !ok {
 		return nil, fmt.Errorf("missing proposal for slot %d", slot)
 	}
 
-	return &validator.publicKey, nil
+	if v, ok := validator.(ValidatorInfo); ok {
+		return &v.publicKey, nil
+	} else {
+		return &types.PublicKey{}, fmt.Errorf("invalid value stored in cache")
```

Comment:
```suggestion
		return &types.PublicKey{}, errors.New("invalid value stored in cache")
```

Even better, store it in variable and reuse it for `fetchExecutionHash`

Reply:
thanks for the suggestion, added a error value for this